### PR TITLE
Staden IO package

### PR DESCRIPTION
**Pull request by maarten-k**
:information_source:
Bioconda has finished the [GCC7 migration](https://github.com/bioconda/bioconda-recipes/issues/13578). If you are dealing with C/C++ or Python package it can be that you need to rebuild other dependent packages. [Bioconda utils - update-pinning](https://bioconda.github.io/updating.html#updating-recipes-for-a-pinning-change) will assist you with that. If you have any questions please use [issue 13578](https://github.com/bioconda/bioconda-recipes/issues/13578).

----------------

* [x] I have read the [guidelines for bioconda recipes](https://bioconda.github.io/guidelines.html).
* [x] This PR adds a new recipe.
* [x] AFAIK, this recipe **is directly relevant to the biological sciences** (otherwise, please submit to the more general purpose [conda-forge channel](https://conda-forge.org/docs/)).
* [ ] This PR updates an existing recipe.
* [ ] This PR does something else (explain below).

### CODE REVIEW

**Comment by maarten-k** [comment]
This is an update of #7196, which never made it into bioconda.

File: recipes/staden-io-lib/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    run:    
+        - zlib
+        - xz
+        - curl
+        - libdeflate
+
+
+test:
+  commands:
+    - echo
```
Comment (bgruening) [inline recipes/staden-io-lib/meta.yaml]: can you improve the test, for example by useing something like: https://github.com/conda-forge/zlib-feedstock/blob/master/recipe/meta.yaml#L27

File: recipes/staden-io-lib/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+build:
+    number: 0
+    skip: True  # [osx]
+
+
+requirements:
+    build:
+        - {{ compiler('c') }}
+    host:
+
```
Comment (bgruening) [inline recipes/staden-io-lib/meta.yaml]: can you remove the empty lines here and clean the recipe up a little bit?

**Comment by epruesse** [comment]
@biocondabot lint

**Comment by maarten-k** [comment]
@BiocondaBot rebuild

**Comment by epruesse** [comment]
Sorry maarten-k. I broke something I think. :( I'm restarting your build and looking into why the bot won't respond to commands from github issues any more.

**Comment by maarten-k** [comment]
bioconda-test failed and it is not clear to me what went wrong. Can an additional pair of eyes take a look at it?

**Comment by epruesse** [comment]
```

  File "/opt/conda/lib/python3.6/site-packages/conda_build/create_test.py", line 84, in create_shell_files

    f.write(cmd)

TypeError: write() argument must be str, not dict
```

**Comment by maarten-k** [comment]
@epruesse thanks! 

I think the build is not responding again. Can you take a look?

**Comment by epruesse** [comment]
@biocondabot rebuild

**Comment by epruesse** [comment]
You can issue that command as well - it's there for just this purpose. Sometimes CircleCI misses a webhook from github and won't build.

**Comment by epruesse** [comment]
@biocondabot lint

**Comment by epruesse** [comment]
@BiocondaBot merge

**Pull request merged**